benchmarks/src/jmh/java/software/amazon/jdbc/benchmarks/PluginBenchmarks.java

```diff
@@ -138,10 +138,58 @@ public ConnectionWrapper initAndReleaseWithAuroraHostListPlugin() throws SQLExce
     }
   }
 
+  @Benchmark
+  public ConnectionWrapper initAndReleaseWithReadWriteSplittingPlugin() throws SQLException {
+    try (ConnectionWrapper wrapper = new ConnectionWrapper(
+        useReadWriteSplittingPlugin(),
+        CONNECTION_STRING,
+        mockConnectionProvider)) {
+      wrapper.releaseResources();
+      return wrapper;
+    }
+  }
+
+  @Benchmark
+  public ConnectionWrapper initAndReleaseWithAuroraHostListAndReadWriteSplittingPlugin()
+      throws SQLException {
+    try (ConnectionWrapper wrapper = new ConnectionWrapper(
+        useAuroraHostListAndReadWriteSplittingPlugin(),
+        CONNECTION_STRING,
+        mockConnectionProvider)) {
+      wrapper.releaseResources();
+      return wrapper;
+    }
+  }
+
+  @Benchmark
+  public ConnectionWrapper initAndReleaseWithReadWriteSplittingPluginWithReaderLoadBalancing()
+      throws SQLException {
+    try (ConnectionWrapper wrapper = new ConnectionWrapper(
+        useReadWriteSplittingPluginWithReaderLoadBalancing(),
+        CONNECTION_STRING,
+        mockConnectionProvider)) {
+      wrapper.releaseResources();
+      return wrapper;
+    }
+  }
+
+  @Benchmark
+  public ConnectionWrapper
+  initAndReleaseWithAuroraHostListAndReadWriteSplittingPluginWithReaderLoadBalancing()
+      throws SQLException {
+    try (ConnectionWrapper wrapper = new ConnectionWrapper(
```

Comment:
https://blog.avenuecode.com/java-microbenchmarks-with-jmh-part-2#:~:text=Parameterizing%20the%20Benchmark
Apparently JMH benchmarks supports parameterized tests, does it make sense to refactor the tests?

Reply:
discussed with Karen, unable to do